api/v1/server/handlers/metadata/health.go

```diff
@@ -1,41 +1,84 @@
 package metadata
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/labstack/echo/v4"
 
 	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
 )
 
-func (u *MetadataService) LivenessGet(ctx echo.Context, request gen.LivenessGetRequestObject) (gen.LivenessGetResponseObject, error) {
+func (u *MetadataService) collectHealthErrors() []error {
+	errs := []error{}
+
 	if !u.config.APIRepository.Health().IsHealthy() {
-		return nil, fmt.Errorf("api repository is not healthy")
+		errs = append(errs, errors.New("api repository is not healthy"))
 	}
 
 	if !u.config.EngineRepository.Health().IsHealthy() {
-		return nil, fmt.Errorf("engine repository is not healthy")
+		errs = append(errs, errors.New("engine repository is not healthy"))
 	}
 
 	if !u.config.MessageQueue.IsReady() {
-		return nil, fmt.Errorf("task queue is not healthy")
+		errs = append(errs, errors.New("task queue is not healthy"))
 	}
 
-	return gen.LivenessGet200Response{}, nil
+	return errs
 }
 
-func (u *MetadataService) ReadinessGet(ctx echo.Context, request gen.ReadinessGetRequestObject) (gen.ReadinessGetResponseObject, error) {
-	if !u.config.APIRepository.Health().IsHealthy() {
-		return nil, fmt.Errorf("api repository is not healthy")
+func (u *MetadataService) logHealthErrors(errs []error) {
+	for _, err := range errs {
+		u.config.Logger.Err(err).Msg("health check failed")
 	}
+}
 
-	if !u.config.EngineRepository.Health().IsHealthy() {
-		return nil, fmt.Errorf("engine repository is not healthy")
+func (u *MetadataService) LivenessGet(ctx echo.Context, request gen.LivenessGetRequestObject) (gen.LivenessGetResponseObject, error) {
+	allErrors := u.collectHealthErrors()
```

Comment:
Let's also set a context timeout which is shorter than our default readiness/liveness probe timeouts, so we can guarantee that we log something